config/sets/doctrine-dbal-40.php

```diff
@@ -5,14 +5,36 @@
 use Rector\Config\RectorConfig;
 use Rector\Renaming\Rector\ClassConstFetch\RenameClassConstFetchRector;
 use Rector\Renaming\Rector\MethodCall\RenameMethodRector;
+use Rector\Renaming\Rector\Name\RenameClassRector;
 use Rector\Renaming\ValueObject\MethodCallRename;
 use Rector\Renaming\ValueObject\RenameClassAndConstFetch;
 
 return static function (RectorConfig $rectorConfig): void {
-    // @see https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#bc-break-removed-connectionparam__array-constants
+    $rectorConfig->ruleWithConfiguration(RenameMethodRector::class, [
+        // @see https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#bc-break-removed-misspelled-isfullfilledby-method
+        new MethodCallRename('Doctrine\DBAL\Schema\Index', 'isFullfilledBy', 'isFulfilledBy'),
+        // @see https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#bc-break-removed-expressionbuilder-methods
+        new MethodCallRename('Doctrine\DBAL\Query\Expression\ExpressionBuilder', 'andX', 'and'),
+        new MethodCallRename('Doctrine\DBAL\Query\Expression\ExpressionBuilder', 'orX', 'or'),
+        // @see https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#bc-break-removed-compositeexpression-methods
+        new MethodCallRename('Doctrine\DBAL\Query\Expression\CompositeExpression', 'add', 'with'),
+        new MethodCallRename('Doctrine\DBAL\Query\Expression\CompositeExpression', 'addMultiple', 'with'),
```

Comment:
Not sure if this is a valid replacement also when i check https://github.com/doctrine/dbal/pull/3858/files#diff-829b37fd7ed65c7ead12002f3e689f3a672986dd88fd0f5d4d5f92321a61211dR1042 i looks like we need to add ``with($parts)` as `with(...$parts)`

Reply:
that seems need a special rule for changing $parts to ...$parts.

Reply:
I added rule ChangeCompositeExpressionAddMultipleWithWithRector but i dont know how to add `...` and `$parts` (argument) together like `...$parts` as a argument do you know?

Reply:
You may can look how to use it with ArrayItem on ArraySpreadInsteadOfArrayMergeRector:

https://github.com/rectorphp/rector-src/blob/e8dd953f10a7afc3610d5185c6b42216e6d9e050/rules/Php74/Rector/FuncCall/ArraySpreadInsteadOfArrayMergeRector.php#L189-L192

Reply:
Thanks that helped me solve it :+1: